test/xUnit/csharp/test_Subsystem.cs

```diff
@@ -30,26 +29,66 @@ private static void VerifySubsystemMetadata(SubsystemInfo ssInfo)
             Assert.Empty(ssInfo.RequiredFunctions);
         }
 
+        private static void VerifyCrossPlatformDscMetadata(SubsystemInfo ssInfo)
+        {
+            Assert.Equal(SubsystemKind.CrossPlatformDsc, ssInfo.Kind);
+            Assert.Equal(typeof(ICrossPlatformDsc), ssInfo.SubsystemType);
+            Assert.True(ssInfo.AllowUnregistration);
+            Assert.False(ssInfo.AllowMultipleRegistration);
+            Assert.Empty(ssInfo.RequiredCmdlets);
+            Assert.Empty(ssInfo.RequiredFunctions);
+        }
+
+        private static void VerifyAllSubsystemInfo(ReadOnlyCollection<SubsystemInfo> ssInfos, SubsystemInfo predictorInfo, SubsystemInfo crossPlatformDscInfo)
+        {
+            Assert.Equal(2, ssInfos.Count);
+            foreach (SubsystemInfo si in ssInfos)
+            {
+                if (si.SubsystemType == typeof(ICommandPredictor))
+                {
+                    Assert.Same(si, predictorInfo);
+                }
+
+                if (si.SubsystemType == typeof(ICrossPlatformDsc))
+                {
+                    Assert.Same(si, crossPlatformDscInfo);
+                }
+            }
+        }
+
         [Fact]
         public static void GetSubsystemInfo()
         {
-            SubsystemInfo ssInfo = SubsystemManager.GetSubsystemInfo(typeof(ICommandPredictor));
+            SubsystemInfo predictorInfo = SubsystemManager.GetSubsystemInfo(typeof(ICommandPredictor));
 
-            VerifySubsystemMetadata(ssInfo);
-            Assert.False(ssInfo.IsRegistered);
-            Assert.Empty(ssInfo.Implementations);
+            VerifyCommandPredictorMetadata(predictorInfo);
+            Assert.False(predictorInfo.IsRegistered);
+            Assert.Empty(predictorInfo.Implementations);
+
+            SubsystemInfo predictorInfo2 = SubsystemManager.GetSubsystemInfo(SubsystemKind.CommandPredictor);
+            Assert.Same(predictorInfo2, predictorInfo);
+
+            SubsystemInfo crossPlatformDscInfo = SubsystemManager.GetSubsystemInfo(typeof(ICrossPlatformDsc));
+
+            VerifyCrossPlatformDscMetadata(crossPlatformDscInfo);
+            Assert.False(crossPlatformDscInfo.IsRegistered);
+            Assert.Empty(crossPlatformDscInfo.Implementations);
 
-            SubsystemInfo ssInfo2 = SubsystemManager.GetSubsystemInfo(SubsystemKind.CommandPredictor);
-            Assert.Same(ssInfo2, ssInfo);
+            SubsystemInfo crossPlatformDscInfo2 = SubsystemManager.GetSubsystemInfo(SubsystemKind.CrossPlatformDsc);
+            Assert.Same(crossPlatformDscInfo2, crossPlatformDscInfo);
 
             ReadOnlyCollection<SubsystemInfo> ssInfos = SubsystemManager.GetAllSubsystemInfo();
-            Assert.Single(ssInfos);
-            Assert.Same(ssInfos[0], ssInfo);
+            VerifyAllSubsystemInfo(ssInfos, predictorInfo, crossPlatformDscInfo);
```

Comment:
Instead of having a method, let's directly do the checks here like this:
```
Assert.Equal(2, ssInfos.Count);
Assert.Same(ssInfos[0], predictorInfo);
Assert.Same(ssInfos[1], crossPlatformDscInfo);
```
The order is the same as how it's declared in the `SubsystemManager` type, so we can safely use `ssInfos[0]` and `ssInfos[1]` here.

Reply:
updated.